Difference between 1.x & 2.x

zoonman opened the pull request.

tbetbetbe committed:
Merge pull request #599 from localheinz/fix/install

Fix: Install dependencies in install section on Travis

tbetbetbe committed:
Merge pull request #600 from localheinz/feature/cache

Enhancement: Cache dependencies between builds

localheinz committed:
Fix: Remove ComputeAccounts

localheinz committed:
Fix: Update Composer itself in before_install section

localheinz committed:
Enhancement: Require squizlabs/php_codesniffer as a dev dependency

localheinz committed:
Enhancement: Run phpcs on Travis

localheinz committed:
Enhancement: Run phpcs only on PHP 5.6

localheinz committed:
Enhancement: Finish early

localheinz committed:
Fix: Run phpcbf to fix issues

localheinz committed:
Fix: Inline autoload function to fix non-camel-caps issue

localheinz committed:
Fix: Update README.md

localheinz committed:
Fix: Run phpcbf again

tbetbetbe committed:
Merge pull request #604 from localheinz/feature/phpcs

Enhancement: Run phpcs on Travis

tbetbetbe committed:
Merge pull request #601 from localheinz/fix/accounts

Fix: Remove ComputeAccounts in favour of Computeaccounts

jeanpaulgalea committed:
Stricter unix permissions for cache files.

- only allow us (user running the code) to read/write from cache files.

	i.e. chmod 0700 on directories
	i.e. chmod 0600 on files

- this is especially important when caching auth tokens,
	where previously it was world readable by anyone on the same machine:

	jean@machine:~# ls -l /tmp/Google_Client/5a/
	total 4
	-rw-r--r-- 1 jean jean 150 Apr 30 22:00 5ac38a276884f502e08b2d0e7dffb128

eRadical committed:
+ Make IO/Stream.php be App Engine aware (The issues were discussed in Case #608 and in Case #570)

bshaffer committed:
updates pubsub from v1beta2 to v1

tbetbetbe committed:
Merge pull request #637 from bshaffer/upgrade-pubsub-to-v1

updates pubsub from v1beta2 to v1

bshaffer committed:
allows fetching/storing of access tokens for testing without modifying git-controlled files

tbetbetbe committed:
Merge pull request #602 from localheinz/fix/self-update

Fix: Update Composer itself in before_install section

tbetbetbe committed:
Merge pull request #643 from bshaffer/easier-access-token-testing

easier access token testing

bshaffer committed:
Refactor for adding UsableAuth library and GuzzleHttp - Phase 1

eRadical committed:
* Leave the unrelying curl library decide what SSL version to use. As discussed in case #639.

itsazzad committed:
remove space typo from ['upload_token ']

crishoj committed:
Fail fast if JSON encoding fails

Let the `call` method fail fast in case `json_encode` fails to encode the request body, rather than proceeding with invalid data.

bshaffer committed:
updates Drive API to add orderBy & other params

tbetbetbe committed:
Merge pull request #666 from bshaffer/issue-656-update-drive-api

[The PR of the Beast] updates Drive API to add orderBy & other params

dmytro-dymarchuk committed:
Google_Http_MediaFileUpload::chunkSize make settable

bshaffer committed:
test refactoring
 * test files/classes match class names
 * removes tests/general/AuthTest.php because it tested firebase-jwt library, and not this library.
 * removes tests/testdata/*.pem, tests/testdata/*.p12, and tests/testdata/*.json as they are no longer used.

tillz committed:
Add methods to resume Uploads

Renamed the 'getResumeUri'-Method to 'fetchResumeUri', as it's more appropriate and 'getResumeUri' is now used as a getter for the private resumeUri property. Added a 'resume'-method to be called with an resumeIri, which will fetch the upload status & setup the object. private makePutRequest method is added to avoid duplicating code from nextChunk().

bshaffer committed:
Merge pull request #672 from dmitrydymarchuk/media-file-upload-chunksize-make-settable

Google_Http_MediaFileUpload::chunkSize make settable

bshaffer committed:
Removes Http, Logger, and Auth

logger and cache cleanup, autoload cleanup

bshaffer committed:
fix #669 - update Bigquery for UDF

bshaffer committed:
adds Google Apps Script Execution API

bshaffer committed:
Merge pull request #683 from bshaffer/add-google-apps-script-execution-api

adds Google Apps Script Execution API

bshaffer committed:
Merge pull request #677 from bshaffer/issue-669

fix #669 - update Bigquery for UDF

bshaffer committed:
DX, organization, and auth cleanup

bshaffer committed:
Merge pull request #617 from jeanpaulgalea/unixperm

Stricter unix permissions for cache files.

bshaffer committed:
Merge pull request #646 from itsazzad/patch-1

remove space typo from ['upload_token ']

bshaffer committed:
Merge pull request #675 from tillz/patch-1

Add methods to resume Uploads

bshaffer committed:
adds UPGRADING.md

bshaffer committed:
This change has been generated by a script that has detected changes in the

discovery doc of the API.

Check http://developers.google.com/discovery for more info.

bshaffer committed:
removes bind_direct for now

bshaffer committed:
Merge pull request #693 from bshaffer/update-generated-libs

Library Update 1.2.0

Commit:
Accept multiple issuers
 - support issuers accounts.google.com and https://accounts.google.com/ by default.

bshaffer committed:
adds tests + fixes bugs for multiple issuers

bshaffer committed:
makes new tests for verifyIdToken php-53-array friendly

bshaffer committed:
updates from master + alpha testers

bshaffer committed:
Merge pull request #695 from bshaffer/v2-rcb-1

v2.0 RCB-1

bshaffer committed:
This change has been generated by a script that has detected changes in the
discovery doc of the API.

Check http://developers.google.com/discovery for more info.

bshaffer committed:
Merge pull request #704 from bshaffer/update-generated-libs

Update v2 generated libs

sqrrrl committed:
Fix #709 - Don't attempt JSON parsing if alt=media requested

bshaffer committed:
Merge pull request #710 from sqrrrl/master

Fix #709 - Don't attempt JSON parsing if alt=media requested

bshaffer committed:
Updated Pubsub.php

This change has been generated by a script that has detected changes in the
discovery doc of the API.

Check http://developers.google.com/discovery for more info.

bshaffer committed:
uses guzzle StreamHandler by default for AppEngine

bshaffer committed:
Merge pull request #712 from bshaffer/update-generated-libs-pubsub

Updated Pubsub to v1

sqrrrl committed:
Prefer parameter definitions from API

bshaffer committed:
Merge pull request #721 from sqrrrl/master

Prefer parameter definitions from API

bshaffer committed:
fix fetchAccessTokenWithAssertionCredentials

bshaffer committed:
Merge pull request #715 from bshaffer/switch-to-stream-handler-by-default

uses guzzle StreamHandler by default [AppEngine only]

sqrrrl committed:
Don't try to convert to model if alt=media present

bshaffer committed:
adds composer support in README and adds autoload back

bshaffer committed:
Merge pull request #727 from bshaffer/fix-fetch-access-token-with-assertion-credentials

fix fetchAccessTokenWithAssertionCredentials

bshaffer committed:
adds tests and refactoring for when alt=media

bshaffer committed:
Merge pull request #732 from bshaffer/sqrrrl-master

Additional Support for alt=media

bshaffer committed:
Merge pull request #729 from bshaffer/better-support-for-composer-update

adds composer support in README and adds autoload back

bshaffer committed:
bump google auth to v0.4 and add JWT3.0 support as a result

bshaffer committed:
Merge pull request #733 from bshaffer/bump-google-auth-and-add-jwt-support

bump google auth to v0.4 and add JWT3.0 support as a result

bshaffer committed:
ensures json errors are not thrown

bshaffer committed:
fixes batch requests for POST

bshaffer committed:
Merge pull request #735 from bshaffer/adds-json-error-handling

ensures json errors are not thrown

bshaffer committed:
Merge pull request #737 from bshaffer/fix-batch-for-post

fixes batch requests for POST

bshaffer committed:
removes v for packagist verification

bshaffer committed:
Merge pull request #746 from bshaffer/remove-v-for-composer

removes v for packagist verification

bshaffer committed:
add banner to README to clarify version docs discrepancy

bshaffer committed:
fixes #574 - checks for cachedir trailing slash

bshaffer committed:
Merge pull request #752 from bshaffer/make-version-discrepancy-more-obvious

add banner to README to clarify version docs discrepancy

bshaffer committed:
Merge pull request #753 from bshaffer/issue-574

fixes #574 - checks for cachedir trailing slash

jordikroon committed:
Remove unused parameter $storageFile in unlock method

jordikroon committed:
Remove unlock parameter from usages

bshaffer committed:
force 0.4 for google-auth

bshaffer committed:
Merge pull request #771 from jordi12100/patch-1

Remove unused parameter $storageFile in unlock method

bshaffer committed:
fix examples & cache tests

bshaffer committed:
add support for guzzle 6

bshaffer committed:
Merge pull request #768 from google/add-support-for-guzzle-6

Add support for guzzle 6

bshaffer committed:
fixes batch uploads with PSR7

bshaffer committed:
Merge pull request #789 from bshaffer/fix-786-for-deferred-uploads

fixes batch uploads with PSR7

bshaffer committed:
removes curl constants incase curl extension isnt loaded

bshaffer committed:
Merge pull request #796 from bshaffer/issue-792

removes curl constants incase curl extension isnt loaded

bshaffer committed:
updates UPGRADING to reflect guzzle 6 changes

bshaffer committed:
Merge pull request #797 from bshaffer/add-guzzle-6-to-upgrading

updates UPGRADING to reflect guzzle 6 changes

michael-oro committed:
Corrected return type in phpDoc for Google_Client::fetchAccessTokenWithAssertion().

JamesVanWaza committed:
Update of the Google URL to Github URL

colinfrei committed:
Handle exceptions correctly when using Guzzle 5.

bshaffer committed:
Merge pull request #808 from colinfrei/guzzle5exceptions

Handle exceptions correctly when using Guzzle 5

bshaffer committed:
Merge pull request #807 from jgacuca567/patch-1

Update of the Google URL to Github URL

bshaffer committed:
Merge pull request #800 from mbessolov/master

Corrected return type in phpDoc for Google_Client::fetchAccessTokenWithAssertion().

AM63 committed:
Incorrect @return line for Client::authorize

Was "@return void", whereas the authorize method does return a ClientInferface object in current release.

huebs committed:
Enable custom umask for cache files

While google/google-api-php-client#617 is well intentioned for its
security concerns, it has an annoying side effect when the cache is
shared between an application's CLI and Web Interface.  Unless you use
the same username for Apache or PHP-FPM to access the CLI (which you
shouldn't), a cache file will only be readable by whichever interface
created it.

My solution is to add a third constructor argument to `Google_Cache_File`
for setting a custom umask.  I've assigned it the default value `077` so
that default permissions for files and directories will remain `0600` and
`0700` respectively.

bshaffer committed:
adds 1-second leeway to JWT validation to prevent intermittent exceptions

bshaffer committed:
Merge pull request #828 from google/issue-827-add-leeway

adds 1-second leeway to JWT validation to prevent intermittent except…

bshaffer committed:
updates google services and tests to match

bshaffer committed:
uses umask function instead of custom umask

bshaffer committed:
Merge pull request #839 from google/file-cache-umask

File cache umask

bshaffer committed:
use hashed username in cache dir

bshaffer committed:
addresses root cache dir permissions issue

bshaffer committed:
mention removal of loadServiceAccountJson in UPGRADING.md

bshaffer committed:
removes auth library caching

bshaffer committed:
Merge pull request #840 from google/add-username-to-cache

use hashed username in cache dir

bshaffer committed:
Merge pull request #841 from bshaffer/issue-832

mention removal of loadServiceAccountJson in UPGRADING.md

bshaffer committed:
removes createDefaultCache method and fixes tests

bshaffer committed:
Merge pull request #842 from google/issue-821

removes auth library caching

kunit committed:
use correct get process user function

kunit committed:
get process user functions return empty, throws exception

bshaffer committed:
phpsec workaround

kunit committed:
remove exception at functions return empty

bshaffer committed:
Merge pull request #848 from google/phpinfo-workaround-for-mvms

phpsec workaround

bshaffer committed:
Merge pull request #847 from kunit/file-cache-dir-owner

use correct get process user function

itsazzad committed:
removed unused line

```
$service = new Google_Service_Drive($client);
```
This line is not seems to be necessary here.

bshaffer committed:
Merge pull request #835 from bshaffer/update-services

updates google services and tests to match

bshaffer committed:
Merge pull request #857 from itsazzad/patch-3

removed unused line

itsazzad committed:
id_token_token instead of id_token

Should it not be so?

itsazzad committed:
multi-api-token instead of access_token

Should it not be so?

bshaffer committed:
Merge pull request #858 from itsazzad/patch-4

id_token_token instead of id_token

bshaffer committed:
Merge pull request #859 from itsazzad/patch-5

multi-api-token instead of access_token

bshaffer committed:
header lines get messed up without this, although it doesn't seem to matter

bshaffer committed:
fixes undefined buildPsr7Response bug (#864)

bshaffer committed:
adds default memory cache, and removes redundant middlewares

bshaffer committed:
Merge pull request #867 from bshaffer/add-memory-cache-for-auth

adds default memory cache, and removes redundant middlewares

bshaffer committed:
Merge pull request #865 from bshaffer/minor-batch-tweak

header lines get messed up without this

bshaffer committed:
bump google auth to 0.6

bshaffer committed:
Merge pull request #870 from bshaffer/bump-google-auth

bump google auth to 0.6

bshaffer committed:
Merge pull request #866 from bshaffer/issue-864

fixes undefined buildPsr7Response bug (#864)

bshaffer committed:
cleans up REST::decodeResponse

bshaffer committed:
Merge pull request #876 from bshaffer/cleans-up-rest-decode-response

cleans up REST::decodeResponse

bshaffer committed:
fixes namespace and adds test

Commit:
Fix link to Travis-CI in README.md

bshaffer committed:
Merge pull request #881 from swordbeta/master

Fix link to Travis-CI in README.md

bshaffer committed:
Merge pull request #880 from bshaffer/fix-response-namespace-in-rest

fixes namespace and adds test

bshaffer committed:
upgrades google auth version

bshaffer committed:
Merge pull request #882 from bshaffer/upgrade-google-auth

upgrades google auth version

bshaffer committed:
Update Google Services from Discovery

bshaffer committed:
Merge pull request #887 from bshaffer/update-google-services

Update Google Services from Discovery

bshaffer committed:
switches this library to pull in generated classes via composer

bshaffer committed:
Merge pull request #893 from bshaffer/move-services-to-separate-repo

switches this library to pull in generated classes via composer

bshaffer committed:
passes in cache to token verifier by default

bshaffer committed:
Merge pull request #895 from bshaffer/issue-894

passes in cache to token verifier by default

danielmirchandani committed:
ExpiredException didn't have the correct namespace

bshaffer committed:
Create README.md

bshaffer committed:
Merge pull request #934 from google/bshaffer-patch-1

Create README.md in src/Google/Services

bshaffer committed:
adds PSR-6 and removes custom caching (#943)

* adds PSR-6 and removes custom caching

* Adds callback

* documents these changes in the README

bshaffer committed:
Update README for v2.0 stable release (#968)

bshaffer committed:
use google/auth 0.9 (#970)

bshaffer committed:
adds service account, OAuth, general usage, and proxying to README (#972)

bshaffer committed:
adds auth code flow to README (#989)

bshaffer committed:
allows string tokens in revoke method (#993)

bshaffer committed:
tag client services (#1000)

Commit:
Fix incorrect function call in UPGRADING.md (#996)

bshaffer committed:
add issue template and link to CONTRIBUTING.md in README (#1003)

bshaffer committed:
bump apiclient-services version (#1006)

bshaffer committed:
updates auth and uses in-memory cache (#1019)

holtkamp committed:
Remove references to Google_Config (#1029)

to prevent confusion

inverse committed:
Update Client.php (#1030)

Correct DocBlock @throws to LogicException

bshaffer committed:
fixes issue with BC breaking google/auth change (#1028)

* fixes issue with BC breaking google/auth change

bshaffer committed:
uses syslog handler on appengine (#1025)

* uses syslog handler on appengine

mortonfox committed:
Fix issue tracker link (#1039)

bocharsky-bw committed:
Highlight class constant in docs on GitHub (#1038)

bshaffer committed:
updates google/apiclient-services to 0.7 (#1053)

bshaffer committed:
fixes cached client tests (#1079)

nathanmarcos committed:
Fix the approval_prompt default value. (#1066)

bshaffer committed:
updates composer dependencies (#1082)

* updates composer dependencies

bshaffer committed:
removes stash logic, adds caching test for MemoryCache (#1085)

SDKiller committed:
Fixed misnamed param in phpdoc (#1107)

bshaffer committed:
fixes issue 1087 - ensures created is set for fetchAccessTokenWithAssertion (#1113)

danielemontesi committed:
Fix unused params in BaseTest (#1091)

bshaffer committed:
allows for older versions of phpseclib (#1118)

* allows for older versions of phpseclib
* uses --prefer-lowest instead of explicitly setting guzzle versions
* remove hhvm, add php 7.1

bshaffer committed:
Update README.md (#1114)

danielemontesi committed:
Remove unused $authHttp to Google_Client::authorize (#1090)

bshaffer committed:
update libver version (#1120)

bshaffer committed:
ensures refresh token is added back to the access token, adds tests (#1121)

bshaffer committed:
ensures JWT leeway is restored after we set it (#1122)

bshaffer committed:
fixes reference to jwt config

Commit:
Don't distribute tests, build files etc. for composer (#1069)

Don't distribute tests, build files etc. when installing with Composer with `--prefer-dist` (the default). This saves lots of bandwith and file extraction when installing dependencies with default settings.

maqnouch committed:
Fix typos in README(#1081)

bshaffer committed:
updates Google_Client::LIBVER to 2.1.1

bshaffer committed:
removes errant symlinks

bshaffer committed:
adds Large File Download example (#1127)

bshaffer committed:
update google/apiclient-servies to v0.9 (#1128)

Big-Shark committed:
Added @deprecated comments (#1132)

IDLOCKED committed:
Fix typos in README (#1130)

bshaffer committed:
fix createBatch call (#1137)

bshaffer committed:
upgrade php client services to 0.11 (#1171)

kalessil committed:
Static Code Analysis with Php Inspections (EA Extended) (#1174)

* language level fixes
* dead code dropped
* one-time use variables inlined
* control flow simplifications
* changes code review
* unnecessary parenthesises dropped
* fixed CS

ManInTheBox committed:
Fixed logger property docblock. (#1177)

Logger is an implementation of `Psr\Log\LoggerInterface`.

bshaffer committed:
Improve model contructor api (#1191)

* improves Model construction by allowing nested properties through array passing
* adds support for passing in created instances of properties in constructor

bshaffer committed:
Michaelbausor fix url shortener sample (#1240)

bshaffer committed:
bumps to latest version of auth client and services (#1242)

bshaffer committed:
bumps minimum guzzle version to 5.3.1 for php7.1 compatibility (#1241)

Commit:
Added check for if range exists when getting progress for resumable upload. (#1209)

amosmos committed:
fetchAccessTokenWithAuthCode already sets the token (#1251)

bshaffer committed:
allows for firebase 5.0 (#1258)

danoscarmike committed:
Add notice about maintenance mode (#1259)

bshaffer committed:
allow for any pre-1.0 version of apiclient-services (#1260)

bshaffer committed:
update LIBVER constant (#1243)

sypherlev committed:
Added missing is_array check (#1249)

ricardofontanelli committed:
Fix directory and reports stop channel urls

Report stop channel and Directory stop channel methods uses a different service path then their default class. When the lib build the full service url, the service path is duplicated.

ricardofontanelli committed:
Remove white spaces at the end of the lines 258 and 262

ricardofontanelli committed:
Fix a requestUrl composition according the pattern used by google-api-php-client-services and provide a test to ensure that all changes doesn't break things

GasparMDQ committed:
Add missing exception to the catch chain (#1284)

kinow committed:
Add missing space to README (#1282)

bshaffer committed:
addresse keytype conflict (#1310)

bshaffer committed:
increment Google_Client::LIBVER to 2.2.1 (#1311)

carusogabriel committed:
Use PHPUnit\Framework\TestCase instead of PHPUnit_Framework_TestCase (#1336)

bshaffer committed:
Add PHP 7.2 in preparation of the release tomorrow (#1351)

carusogabriel committed:
Refactoring tests (#1352)

deviantintegral committed:
Document that not all APIs support service accounts

redcapital committed:
fix certificate caching, fixes #1033 (#1322)

carusogabriel committed:
Refactoring tests (#1355)

ajmichels commented:
This PR probably isn't necessary anymore right?

danieljpostVivial commented:
I'm thinking of this repo as deprecated but not removed. I suspect there's more "custom" code in here that we haven't carried over into the new repo.

michaelbausor committed:
Remove beta

Phobetor committed:
Fix method documentation in Client.php (#1317)

bshaffer committed:
Fixes guzzle5 tests (#1378)

ricardofontanelli committed:
Update ResourceTest.php

CHANGE: use proper comment blocks

mattwhisenhunt committed:
Promote GCP APIs (#1416)

JustinBeckwith committed:
Merge pull request #1354 from deviantintegral/service-account-docs

Document that not all APIs support service accounts

bshaffer committed:
ensures leeway is at least 1, instead of setting it to 1 (#1455)

mattwhisenhunt committed:
Update readme.md, remove Stash, add Cache (#1458)

fixes #1101

bshaffer committed:
increment libver to 2.2.2 (#1462)

mattwhisenhunt committed:
examples: Remove extra setAccessKey call. (#1469)

* examples: Remove extra setAccessKey call.

* examples: fix broken redirect

theacodes committed:
Add Code of Conduct

NanneHuiges committed:
Updates phpdoc for Google_Client::setScopes (#1465)

Assimilationstheorie committed:
Small changes to examples/index.php (#1460)

Assimilationstheorie committed:
misc updates to examples/url-shortener.php (#1456)

Yahav committed:
Fixing the ability to provide custom retryMap (#1488)

Fixes #1487

siburny committed:
Update docs to install latest 2.x version (#1491)

Escape caret symbol properly

chingor13 committed:
Use new auth urls (#1492)

mattwhisenhunt committed:
Update default and doc values for setApprovalPrompt. (#1535)

* Update default and doc values for setApprovalPrompt.

JustinBeckwith committed:
Update GitHub issue templates

mattwhisenhunt committed:
Update README.md (#1547)

ddbruce committed:
Fix build status image (#1549)

dwsupplee committed:
fix tests (#1560)

mattwhisenhunt committed:
Revert #1535. (#1556)

adamsilverstein committed:
Use double pipes in composer dependency file (#1505)

use of single pipes is deprecated, see https://github.com/composer/composer/issues/6755#issue-266478234

jdpedrie committed:
Add UnexpectedValueException to documentation (#1592)

szepeviktor committed:
Update `Google_Client::revokeToken()` docblock. (#1609)

as Google_AccessToken_Revoke::revokeToken() accepts it

jdpedrie committed:
Add batch reset warning (#1613)

m6w6 committed:
Fix usage of undefined constants (#1621)

gpenverne committed:
Specify configuration filepath  when config file is not found (#1626)

jdpedrie committed:
Add guzzle config to readme (#1625)

bshaffer committed:
Prepare for v2.2.3 (#1633)

grant committed:
Delete url-shortener.php (#1646)
